modules/management-console-agent/src/main/java/org/apache/ignite/internal/agent/action/controller/QueryActionsController.java

```diff
@@ -89,6 +89,19 @@ public QueryActionsController(GridKernalContext ctx) {
         return new IgniteFinishedFutureImpl<>();
     }
 
+    /**
+     * @param globalQryId Global query id.
+     * @return Kill query result.
+     */
+    public IgniteInternalFuture<?> kill(String globalQryId) {
+        return ctx.closure().runLocalSafe(() -> {
```

Comment:
Should we have checks features that "KILL QUERY" is supported by node?

Reply:
May be we should define some interceptor for all actions that can validate feature/version from annotation on action method?

Reply:
Why we should check it on the agent? I think we should check that we can run some action on UI as we do it with baseline auto-adjust.